### PR TITLE
Improve MultiprocessIterator

### PR DESCRIPTION
**Pull request by okuta**
- Avoid ctrl+C problem
  + https://stackoverflow.com/questions/1408356/keyboard-interrupts-with-pythons-multiprocessing-pool
  + https://stackoverflow.com/questions/11312525/catch-ctrlc-sigint-and-exit-multiprocesses-gracefully-in-python
- Fix `__copy__` to improve readability.
- Use list comprehension instead of `list` function.

### CODE REVIEW

File: chainer/iterators/multiprocess_iterator.py [hunk 1]
```diff
@@ -338,6 +342,7 @@ def _proceed(self):
 
 def _fetch_setup(dataset, mem_size, mem_bulk):
     global _fetch_dataset, _fetch_mem_size, _fetch_mem_bulk
+    signal.signal(signal.SIGINT, signal.SIG_IGN)
```
Comment (grafi-tt) [inline chainer/iterators/multiprocess_iterator.py:345]: Is this change intended to not mess up stacktrace?

Comment (grafi-tt) [reply chainer/iterators/multiprocess_iterator.py:345]: Sorrry, I didn't see your explanation.

Comment (okuta) [reply chainer/iterators/multiprocess_iterator.py:345]: Do not mind.

**Comment by okuta** [comment]
Thank you!
